### PR TITLE
Allow to configure the list of ElasticSearch indices

### LINKED ISSUE #5978: Allow to configure the list of ElasticSearch indices (to be able to add custom ones)
## What is the motivation for adding / enhancing this feature?

For big custom features we might need to add new ElasticSearch indices. And to be able to make requests into those new indices, we need to be able to add those indices to the SearchAdapter class.

Yes, we can technically do that by calling `registerEntityType` method, but there doesn't seem to be a proper place to do that. I feel like it would be better if it was possible to customise the available Elastic indices list through config file.

## What are the acceptance criteria 

- [x] Elastic search indices are customisable and can be changed in configuration file

## Version of Vue Storefront

- [x] Vue Storefront
- [ ] Vue Storefront Next

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by SergeyDarnopykh**
### Related Issues

closes #5978

### Short Description of the PR
Allow to configure the list of ElasticSearch indices (to be able to add custom ones)

### Pull Request Checklist

- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [x] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [x] Default Theme
- [x] Capybara Theme
- [x] I have written test cases for my code

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -121,6 +121,7 @@
     "httpAuth": "",
     "host": "/api/catalog",
     "index": "vue_storefront_catalog",
+    "indices": ["product", "attribute", "category", "taxrule", "review", "cms_page", "cms_block", "cms_hierarchy"],
```
Comment (lukaszjedrasik) [inline config/default.json:124]:
```suggestion
    "indices": [
       "product", 
       "attribute", 
       "category", 
       "taxrule", 
       "review", 
       "cms_page", 
       "cms_block", 
       "cms_hierarchy"
     ],
```

File: docs/guide/data/entity-types.md [hunk 2]
```diff
@@ -6,11 +6,25 @@ Vue Storefront uses multiple data-entity types to cover the whole scope of the s
...
 These entity types were hardcoded and there was no ability to easily use another custom entity type required for customization.
 
 Now, Vue Storefront has a new logic to work with entities in the data-fetching perspective: Entity Types.
 
+Also, the list of used elastic indices (Entity types) can be customized via `indices` field in the config:
+```json
+ "elasticsearch": {
+    "httpAuth": "",
+    "host": "/api/catalog",
+    "indices": ["product", "attribute", "category", "taxrule", "review", "cms_page", "cms_block", "cms_hierarchy"],
```
Comment (lukaszjedrasik) [inline docs/guide/data/entity-types.md:23]:
```suggestion
    "indices": [
       "product", 
       "attribute", 
       "category", 
       "taxrule", 
       "review", 
       "cms_page", 
       "cms_block", 
       "cms_hierarchy"
     ],
```

**Review by lukaszjedrasik — CHANGES_REQUESTED** [review]

File: config/default.json (hunk 1 above)
Comment (SergeyDarnopykh) [reply config/default.json]: ✅

File: docs/guide/data/entity-types.md (hunk 2 above)
Comment (SergeyDarnopykh) [reply docs/guide/data/entity-types.md]: ✅

**Review by lukaszjedrasik — APPROVED** [review]

File: config/default.json [hunk 3]
```diff
@@ -121,6 +121,16 @@
     "httpAuth": "",
     "host": "/api/catalog",
     "index": "vue_storefront_catalog",
+    "indices": [
```
Comment (Fifciu) [inline config/default.json:124]: Let's call them `entityTypes`

**Review by Fifciu — CHANGES_REQUESTED** [review]
Great job. But let's rename `indices` to `entityTypes`

File: config/default.json (hunk 3 above)
Comment (SergeyDarnopykh) [reply config/default.json]: ✅

**Review by Fifciu — CHANGES_REQUESTED** [review]
Looks great, now please add info about necessity of adding `entityTypes` property to the Upgrade notes

**Comment by SergeyDarnopykh** [comment]
> Looks great, now please add info about necessity of adding `entityTypes` property to the Upgrade notes

I'm not 100% that's what you meant, but I added the information about elastic entity types to `docs/guide/upgrade-notes/README.md`  into `1.12 -> 1.13` section. If it's not what you meant, let me know and I will change it.

**Review by Fifciu — CHANGES_REQUESTED** [review]
We've just synced develop with 1.12.3 after patch release. Please resolve conflict, then I can merge

**Review by Fifciu — APPROVED** [review]

**Comment by Fifciu** [comment]
Great job, thanks

**Pull request merged**